alonet/detr_panoptic/detr_panoptic.py

```diff
@@ -0,0 +1,222 @@
+Panoptic module to predict object segmentation.
+"""
+import os
+from typing import Callable, Dict
+
+import torch
+import torch.nn.functional as F
+from torch import nn
+
+from alonet.detr.detr_r50 import DetrR50
+from alonet.detr_panoptic.nn import FPNstyleCNN
+from alonet.detr_panoptic.nn import MHAttentionMap
+from alonet.detr_panoptic.utils import get_mask_queries
+from alonet.detr.misc import assert_and_export_onnx
+
+import aloscene
+import alonet
+
+
+class PanopticHead(nn.Module):
+    """Head Pytorch module to predict segmentation masks from previous boxes detection task.
+
+    Parameters
+    ----------
+    DETR_module : alonet.detr.Detr
+        Object detection module based on DETR architecture
+    freeze_detr : bool, optional
+        Freeze DETR_module weights in train procedure, by default True
+    aux_loss: bool, optional
+        Return aux outputs in forward step (if required), by default use DETR_module.aux_loss attribute value
+    device : torch.device, optional
+        Configure module in CPU or GPU, by default torch.device("cpu")
+    weights : str, optional
+        Load weights from name project, by default None
+    """
+
+    INPUT_MEAN_STD = alonet.detr.detr.INPUT_MEAN_STD
+
+    def __init__(
+        self,
+        DETR_module: alonet.detr.Detr,
+        freeze_detr: bool = True,
+        aux_loss: bool = None,
+        device: torch.device = torch.device("cpu"),
+        weights: str = None,
+    ):
+        super().__init__()
+        self.detr = DETR_module
+
+        # Get complement outputs in forward
+        self.detr.return_dec_outputs = True
+        self.detr.return_enc_outputs = True
+        self.detr.return_bb_outputs = True
+        self.detr.aux_loss = aux_loss or self.detr.aux_loss
+
+        # Freeze DETR parameters to not train them
+        if freeze_detr:
+            for p in self.parameters():
+                p.requires_grad_(False)
+
+        # Use values provides of DETR module in MHA and FPN
+        hidden_dim, nheads = DETR_module.transformer.d_model, DETR_module.transformer.nhead
+        self.bbox_attention = MHAttentionMap(hidden_dim, hidden_dim, nheads, dropout=0.1)
+        self.mask_head = FPNstyleCNN(hidden_dim + nheads, [1024, 512, 256], hidden_dim)
+
+        if device is not None:
+            self.to(device)
+
+        self.device = device
+
+        # Load weights
+        if weights is not None:
+            if ".pth" in weights or ".ckpt" in weights:
+                alonet.common.load_weights(self, weights, device, strict_load_weights=True)
+            else:
+                raise ValueError(f"Unknown weights: '{weights}'")
+
+    @assert_and_export_onnx(check_mean_std=True, input_mean_std=INPUT_MEAN_STD)
+    def forward(self, frames: aloscene.frame, get_filter_fn: Callable = None, **kwargs):
+        """PanopticHead forward, that joint to the previous boxes predictions the new masks feature.
+
+        Parameters
+        ----------
+        frames : aloscene.frame
```

Comment:
`get_filter_fn` seem important to described

Reply:
Done